packages/server-core/src/projects/project/project.class.ts

```diff
@@ -391,21 +389,26 @@ export class ProjectService<T = ProjectType, ServiceParams extends Params = Proj
 
       if (!params.user!.scopes?.find((scope) => scope.type === 'admin:admin'))
         params.query.id = { $in: [...new Set(allowedProjects.map((project) => project.id))] }
-      delete params.query.allowed
-
-      populateProjectPermissions = true
     }
 
-    params = {
+    let paramsWithoutExtras = {
       ...params,
+      // Explicitly cloned sort object because otherwise it was affecting default params object as well.
+      query: params?.query ? JSON.parse(JSON.stringify(params?.query)) : {}
+    }
+
+    paramsWithoutExtras = {
+      ...paramsWithoutExtras,
       query: {
-        ...params?.query,
+        ...paramsWithoutExtras.query,
         $limit: params?.query?.$limit || 1000,
         $sort: params?.query?.$sort || { name: 1 }
       }
     }
 
-    const data = ((await super._find(params)) as Paginated<ProjectType>).data
+    if (paramsWithoutExtras?.query?.allowed) delete paramsWithoutExtras.query.allowed
+
+    const data = ((await super._find(paramsWithoutExtras)) as Paginated<ProjectType>).data
```

Comment:
We also need to ensure `paginate: false` is not being passed here, else the conversion to `Paginated<ProjectType>` will create issues.

We need to discuss why find method is always returning array? The behavior of this service is other than all other services that we have implemented, so it can create confusion.